### DISCUSSION TITLE
Any plans for svelte support?

### DISCUSSION THREAD

**Discussion by borsemayur2** [alephjs/aleph.js#192] [Q&A]
Svelte support would be a great feature.

**Comment by ije** [comment]
i have a plan to support vue as default, svelte i'm not sure, but you can write your own svelte framework(bootstrap.ts for csr, and renderer.ts for ssr) is not difficult, i will update the docs later.

**Reply by calebeaires** [reply]
Vue 3 would be great. Cant wait for the launch!

**Reply by aadamsx** [reply]
I thought React was default?  What does it mean to be Default?

**Reply by calebeaires** [reply]
This is what I understand: multiple built int support: React, Preact, Vue

**Reply by shadowtime2000** [reply]
Basically yeah, support Vue built in along with React. We are also thinking of just making the lifecycle hookable so you can really add any framework.

**Reply by aadamsx** [reply]
> We are also thinking of just making the lifecycle hookable so you can really add any framework.

I like this, you should abstract out the UI framework.  Also from a UI perspective we should be able to hook into what services we like.  Want SSR, there's a hook for that, want SSG okay use that hook, want both (ISR) use the hooks you want from both SSR and SSG.

**Reply by aadamsx** [reply]
By the way, do you have a diagram of the lifecycle for me to review/study?  I'm trying to get down to the metal on Aleph

**Reply by calebeaires** [reply]
This will be a game-changer. Aleph is a good project will some really cool features. We need some alternatives to NuxtJs!
